dsa/src/lib.rs

```diff
@@ -27,40 +29,44 @@
 //! # let read_common_parameters = || (BigUint::one(), BigUint::one(), BigUint::one());
 //! # let read_public_component = || BigUint::one();
 //! # let read_private_component = || BigUint::one();
+//! # || -> signature::Result<()> {
 //! let (p, q, g) = read_common_parameters();
-//! let components = Components::from_components(p, q, g);
+//! let components = Components::from_components(p, q, g)?;
 //!
 //! let x = read_public_component();
-//! let verifying_key = VerifyingKey::from_components(components, x);
+//! let verifying_key = VerifyingKey::from_components(components, x)?;
 //!
 //! let y = read_private_component();
-//! let signing_key = SigningKey::from_components(verifying_key, y);
+//! let signing_key = SigningKey::from_components(verifying_key, y)?;
+//!
+//! # Ok(())
+//! # }();
 //! ```
 //!
 
 extern crate alloc;
 
-/// DSA object identifier as defined by RFC-3279, section 2.3.2
-const DSA_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.840.10040.4.1");
-
-pub use crate::{
-    components::Components, sig::Signature, signing_key::SigningKey, verifying_key::VerifyingKey,
+pub use self::{
+    components::Components, sig::Signature, signing_key::SigningKey, size::KeySize,
+    verifying_key::VerifyingKey,
 };
 
 pub use pkcs8;
 pub use signature;
 
-pub mod consts;
-
 use num_bigint::BigUint;
 use pkcs8::spki::ObjectIdentifier;
 
 mod components;
 mod generate;
 mod sig;
 mod signing_key;
+mod size;
 mod verifying_key;
 
+/// DSA object identifier as defined by RFC-3279, section 2.3.2
+const DSA_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.840.10040.4.1");
```

Comment:
How about:

```suggestion
pub const OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.840.10040.4.1");
```

...accessible as `dsa::OID`?

Reply:
Sounds like a nice addition. Added